lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/csv/CsvSerializer.java

```diff
@@ -81,7 +81,8 @@ public void write(Page page, int position, SliceOutput sliceOutput)
                 }
 
                 Slice value = VARCHAR.getSlice(block, position);
-                if (value.indexOfByte(quoteChar) < 0 && (escapeChar == quoteChar || value.indexOfByte(escapeChar) < 0)) {
+                // if escape is zero, escaping is disabled; otherwise, check if value contains quote or escape character
+                if (escapeChar == '\0' || value.indexOfByte(quoteChar) < 0 && (escapeChar == quoteChar || value.indexOfByte(escapeChar) < 0)) {
```

Comment:
Add parentheses so the order of `||` and `&&` are explicit. I assume this should be
```java
if (escapeChar == '\0' || (value.indexOfByte(quoteChar) < 0 && (escapeChar == quoteChar || value.indexOfByte(escapeChar) < 0))) {